### ISSUE TITLE
Bluetooth stopped working after upgrading HA Yellow from 11.4 to 11.5

### ISSUE THREAD

**Issue by brunokc** [home-assistant/operating-system#3180]
### Describe the issue you are experiencing

I've been trying to upgraded my HA Yellow from HAOS 11.4 to 11.5 but it used to just come back to 11.4 after reboot. Today I tried again and it did upgrade. However, upon reboot, Bluetooth was non-functional (Bluetooth was working fine on 11.4):

<img width="262" alt="image" src="https://github.com/home-assistant/operating-system/assets/1355373/a4c599a2-1b37-406f-bd03-a760a3b54b43">

### What operating system image do you use?

yellow (Home Assistant Yellow)

### What version of Home Assistant Operating System is installed?

11.5

### Did you upgrade the Operating System.

Yes

### Steps to reproduce the issue

1. Upgrade from 11.4 to 11.5
2. Observe Bluetooth is not working

### Anything in the Supervisor logs that might be useful for us?

```txt
24-02-16 17:24:50 INFO (MainThread) [supervisor.os.manager] Detect Home Assistant Operating System 11.5 / BootSlot A
24-02-16 17:24:50 INFO (SyncWorker_1) [supervisor.docker.manager] Starting homeassistant
24-02-16 17:24:50 INFO (MainThread) [supervisor.homeassistant.core] Wait until Home Assistant is ready
24-02-16 17:24:51 INFO (MainThread) [supervisor.store.git] Loading add-on /data/addons/core repository
24-02-16 17:24:51 INFO (MainThread) [supervisor.store.git] Loading add-on /data/addons/git/a0d7b954 repository
24-02-16 17:24:51 INFO (MainThread) [supervisor.store.git] Loading add-on /data/addons/git/229fc559 repository
24-02-16 17:24:51 INFO (MainThread) [supervisor.store.git] Loading add-on /data/addons/git/5c53de3b repository
24-02-16 17:24:53 INFO (MainThread) [supervisor.store] Loading add-ons from store: 78 all - 78 new - 0 remove
24-02-16 17:24:53 INFO (MainThread) [supervisor.addons.manager] Found 5 installed add-ons
24-02-16 17:24:53 INFO (MainThread) [supervisor.docker.interface] Attaching to ghcr.io/hassio-addons/zwave-js-ui/aarch64 with version 3.3.2
24-02-16 17:24:53 INFO (MainThread) [supervisor.docker.interface] Attaching to homeassistant/aarch64-addon-silabs-multiprotocol with version 2.4.4
24-02-16 17:24:53 INFO (MainThread) [supervisor.docker.interface] Attaching to ghcr.io/z-wave-me/aarch64-ha-z-wave-me-addon with version 4.1.0
24-02-16 17:24:53 INFO (MainThread) [supervisor.docker.interface] Attaching to ghcr.io/hassio-addons/vscode/aarch64 with version 5.15.0
24-02-16 17:24:53 INFO (MainThread) [supervisor.docker.interface] Attaching to homeassistant/aarch64-addon-ssh with version 9.9.0
24-02-16 17:24:53 INFO (MainThread) [supervisor.backups.manager] Found 89 backup files
24-02-16 17:24:53 INFO (MainThread) [supervisor.discovery] Loaded 2 messages
24-02-16 17:24:53 INFO (MainThread) [supervisor.ingress] Loaded 2 ingress sessions
24-02-16 17:24:53 INFO (MainThread) [supervisor.resolution.check] Starting system checks with state setup
24-02-16 17:24:53 INFO (MainThread) [supervisor.resolution.check] System checks complete
24-02-16 17:24:53 INFO (MainThread) [supervisor.resolution.evaluate] Starting system evaluation with state setup
24-02-16 17:24:53 INFO (MainThread) [supervisor.resolution.evaluate] System evaluation complete
24-02-16 17:24:53 INFO (MainThread) [supervisor.jobs] 'ResolutionFixup.run_autofix' blocked from execution, system is not running - setup
24-02-16 17:24:53 INFO (MainThread) [supervisor.resolution.evaluate] Starting system evaluation with state setup
24-02-16 17:24:53 INFO (MainThread) [supervisor.resolution.evaluate] System evaluation complete
24-02-16 17:24:53 INFO (MainThread) [__main__] Running Supervisor
24-02-16 17:24:53 INFO (MainThread) [supervisor.os.manager] Rauc: A - marked slot kernel.0 as good
24-02-16 17:24:53 INFO (MainThread) [supervisor.addons.manager] Phase 'initialize' starting 0 add-ons
24-02-16 17:24:53 INFO (MainThread) [supervisor.addons.manager] Phase 'system' starting 1 add-ons
24-02-16 17:24:53 INFO (SyncWorker_1) [supervisor.docker.manager] Cleaning addon_a0d7b954_zwavejs2mqtt application
24-02-16 17:24:54 INFO (MainThread) [supervisor.docker.addon] Starting Docker add-on ghcr.io/hassio-addons/zwave-js-ui/aarch64 with version 3.3.2
24-02-16 17:25:00 INFO (MainThread) [supervisor.resolution.evaluate] Starting system evaluation with state startup
24-02-16 17:25:00 INFO (MainThread) [supervisor.resolution.evaluate] System evaluation complete
24-02-16 17:25:00 INFO (MainThread) [supervisor.homeassistant.api] Updated Home Assistant API token
24-02-16 17:25:00 INFO (MainThread) [supervisor.homeassistant.core] Home Assistant Core state changed to NOT_RUNNING
24-02-16 17:25:00 INFO (MainThread) [supervisor.store.git] Update add-on https://github.com/z-wave-me/ha-z-wave-me-addon repository
24-02-16 17:25:00 INFO (MainThread) [supervisor.store.git] Update add-on https://github.com/esphome/home-assistant-addon repository
24-02-16 17:25:00 INFO (MainThread) [supervisor.store.git] Update add-on https://github.com/home-assistant/addons repository
24-02-16 17:25:00 INFO (MainThread) [supervisor.store.git] Update add-on https://github.com/hassio-addons/repository repository
24-02-16 17:25:03 INFO (MainThread) [supervisor.store] Loading add-ons from store: 78 all - 0 new - 0 remove
24-02-16 17:25:03 INFO (MainThread) [supervisor.store] Loading add-ons from store: 78 all - 0 new - 0 remove
24-02-16 17:25:24 INFO (MainThread) [supervisor.addons.manager] Phase 'services' starting 3 add-ons
24-02-16 17:25:24 INFO (SyncWorker_1) [supervisor.docker.manager] Cleaning addon_a0d7b954_vscode application
24-02-16 17:25:25 INFO (MainThread) [supervisor.docker.addon] Starting Docker add-on ghcr.io/hassio-addons/vscode/aarch64 with version 5.15.0
24-02-16 17:25:25 INFO (SyncWorker_2) [supervisor.docker.manager] Cleaning addon_core_ssh application
24-02-16 17:25:26 INFO (MainThread) [supervisor.docker.addon] Starting Docker add-on homeassistant/aarch64-addon-ssh with version 9.9.0
24-02-16 17:25:26 INFO (SyncWorker_7) [supervisor.docker.manager] Cleaning addon_core_silabs_multiprotocol application
24-02-16 17:25:27 INFO (MainThread) [supervisor.docker.addon] Starting Docker add-on homeassistant/aarch64-addon-silabs-multiprotocol with version 2.4.4
24-02-16 17:25:30 WARNING (MainThread) [supervisor.addons.addon] core_silabs_multiprotocol is already running!
24-02-16 17:25:39 INFO (MainThread) [supervisor.api.middleware.security] /network/info access from core_silabs_multiprotocol
24-02-16 17:25:52 INFO (MainThread) [supervisor.homeassistant.core] Home Assistant Core state changed to RUNNING
24-02-16 17:25:52 INFO (MainThread) [supervisor.homeassistant.core] Detect a running Home Assistant instance
24-02-16 17:25:55 INFO (MainThread) [supervisor.core] Skipping start of Home Assistant
24-02-16 17:25:55 INFO (MainThread) [supervisor.addons.manager] Phase 'application' starting 0 add-ons
24-02-16 17:25:55 INFO (MainThread) [supervisor.misc.tasks] All core tasks are scheduled
24-02-16 17:25:55 INFO (MainThread) [supervisor.core] Supervisor is up and running
24-02-16 17:25:55 INFO (MainThread) [supervisor.host.info] Updating local host information
24-02-16 17:25:55 INFO (MainThread) [supervisor.updater] Fetching update data from https://version.home-assistant.io/stable.json
24-02-16 17:25:55 INFO (MainThread) [supervisor.resolution.check] Starting system checks with state running
24-02-16 17:25:55 INFO (MainThread) [supervisor.resolution.checks.base] Run check for multiple_data_disks/system
24-02-16 17:25:55 INFO (MainThread) [supervisor.resolution.checks.base] Run check for no_current_backup/system
24-02-16 17:25:55 INFO (MainThread) [supervisor.resolution.module] Create new suggestion create_full_backup - system / None
24-02-16 17:25:55 INFO (MainThread) [supervisor.resolution.module] Create new issue no_current_backup - system / None
24-02-16 17:25:55 INFO (MainThread) [supervisor.resolution.checks.base] Run check for free_space/system
24-02-16 17:25:55 INFO (MainThread) [supervisor.resolution.checks.base] Run check for docker_config/system
24-02-16 17:25:55 INFO (MainThread) [supervisor.resolution.checks.base] Run check for pwned/addon
24-02-16 17:25:55 INFO (MainThread) [supervisor.resolution.checks.base] Run check for ipv4_connection_problem/system
24-02-16 17:25:55 INFO (MainThread) [supervisor.resolution.checks.base] Run check for trust/supervisor
24-02-16 17:25:55 INFO (MainThread) [supervisor.resolution.checks.base] Run check for dns_server_ipv6_error/dns_server
24-02-16 17:25:55 INFO (MainThread) [supervisor.resolution.checks.base] Run check for dns_server_failed/dns_server
24-02-16 17:25:55 INFO (MainThread) [supervisor.resolution.checks.base] Run check for security/core
24-02-16 17:25:55 INFO (MainThread) [supervisor.resolution.check] System checks complete
24-02-16 17:25:55 INFO (MainThread) [supervisor.resolution.evaluate] Starting system evaluation with state running
24-02-16 17:25:56 INFO (MainThread) [supervisor.resolution.evaluate] System evaluation complete
24-02-16 17:25:56 INFO (MainThread) [supervisor.resolution.fixup] Starting system autofix at state running
24-02-16 17:25:56 INFO (MainThread) [supervisor.resolution.fixup] System autofix complete
24-02-16 17:25:56 INFO (MainThread) [supervisor.host.services] Updating service information
24-02-16 17:25:56 INFO (MainThread) [supervisor.host.network] Updating local network information
24-02-16 17:25:57 INFO (MainThread) [supervisor.host.sound] Updating PulseAudio information
24-02-16 17:25:57 INFO (MainThread) [supervisor.host.manager] Host information reload completed
24-02-16 17:26:09 INFO (MainThread) [supervisor.hardware.monitor] Detecting remove hardware /dev/bsg/0:0:0:0 - None
24-02-16 17:26:09 INFO (MainThread) [supervisor.hardware.monitor] Detecting remove hardware /dev/sda - /dev/disk/by-id/usb-SanDisk_Cruzer_Mini_2005174010188fd35be7-0:0
24-02-16 17:26:09 INFO (MainThread) [supervisor.hardware.monitor] Detecting remove hardware /dev/bus/usb/001/003 - None
24-02-16 17:55:56 INFO (MainThread) [supervisor.homeassistant.api] Updated Home Assistant API token
24-02-16 18:24:53 INFO (MainThread) [supervisor.resolution.check] Starting system checks with state running
24-02-16 18:24:53 INFO (MainThread) [supervisor.resolution.checks.base] Run check for multiple_data_disks/system
24-02-16 18:24:53 INFO (MainThread) [supervisor.resolution.checks.base] Run check for free_space/system
24-02-16 18:24:53 INFO (MainThread) [supervisor.resolution.checks.base] Run check for docker_config/system
24-02-16 18:24:53 INFO (MainThread) [supervisor.resolution.checks.base] Run check for pwned/addon
24-02-16 18:24:53 INFO (MainThread) [supervisor.resolution.checks.base] Run check for ipv4_connection_problem/system
24-02-16 18:24:53 INFO (MainThread) [supervisor.resolution.checks.base] Run check for trust/supervisor
24-02-16 18:24:53 INFO (MainThread) [supervisor.resolution.checks.base] Run check for dns_server_ipv6_error/dns_server
24-02-16 18:24:53 INFO (MainThread) [supervisor.resolution.checks.base] Run check for dns_server_failed/dns_server
24-02-16 18:24:53 INFO (MainThread) [supervisor.resolution.checks.base] Run check for security/core
24-02-16 18:24:53 INFO (MainThread) [supervisor.resolution.check] System checks complete
24-02-16 18:24:53 INFO (MainThread) [supervisor.resolution.evaluate] Starting system evaluation with state running
24-02-16 18:24:53 INFO (MainThread) [supervisor.resolution.evaluate] System evaluation complete
24-02-16 18:24:53 INFO (MainThread) [supervisor.resolution.fixup] Starting system autofix at state running
24-02-16 18:24:53 INFO (MainThread) [supervisor.resolution.fixup] System autofix complete
24-02-16 18:25:56 INFO (MainThread) [supervisor.homeassistant.api] Updated Home Assistant API token
24-02-16 18:55:56 INFO (MainThread) [supervisor.homeassistant.api] Updated Home Assistant API token
```

### Anything in the Host logs that might be useful for us?

```txt
I didn't see anything useful. See "additional information" below for some useful information.
```

## System Information

version | core-2024.1.6
-- | --
installation_type | Home Assistant OS
dev | false
hassio | true
docker | true
user | root
virtualenv | false
python_version | 3.11.6
os_name | Linux
os_version | 6.1.63-haos-raspi
arch | aarch64
timezone | America/Los_Angeles
config_dir | /config

<details><summary>Home Assistant Community Store</summary>

GitHub API | ok
-- | --
GitHub Content | ok
GitHub Web | ok
GitHub API Calls Remaining | 5000
Installed Version | 1.34.0
Stage | running
Available Repositories | 1393
Downloaded Repositories | 8

</details>

<details><summary>Home Assistant Cloud</summary>

logged_in | false
-- | --
can_reach_cert_server | ok
can_reach_cloud_auth | ok
can_reach_cloud | ok

</details>

<details><summary>Home Assistant Supervisor</summary>

host_os | Home Assistant OS 11.5
-- | --
update_channel | stable
supervisor_version | supervisor-2024.01.1
agent_version | 1.6.0
docker_version | 24.0.7
disk_total | 53.4 GB
disk_used | 8.0 GB
healthy | true
supported | true
board | yellow
supervisor_api | ok
version_api | ok
installed_addons | Z-Wave JS UI (3.3.2), Studio Code Server (5.15.0), Terminal & SSH (9.9.0), Silicon Labs Multiprotocol (2.4.4), Z-Wave.Me Add-on (4.1.0)

</details>

<details><summary>Dashboards</summary>

dashboards | 5
-- | --
resources | 4
views | 4
mode | storage

</details>

<details><summary>Recorder</summary>

oldest_recorder_run | February 7, 2024 at 2:02 AM
-- | --
current_recorder_run | February 16, 2024 at 5:24 PM
estimated_db_size | 43.14 MiB
database_engine | sqlite
database_version | 3.41.2

</details>

### Additional information

kernel module hci_uart is loaded:

```
# lsmod | grep hci
hci_uart               98304  0
btqca                  20480  1 hci_uart
btrtl                  20480  1 hci_uart
btbcm                  24576  1 hci_uart
btintel                40960  1 hci_uart
bluetooth             593920  14 btrtl,btqca,btsdio,btintel,hci_uart,btbcm,bnep

# ls -l /dev/tty[AS]* /dev/serial*
lrwxrwxrwx    1 root     root             7 Feb 17 01:24 /dev/serial0 -> ttyAMA0
crw-rw----    1 root     dialout   204,  64 Feb 17 03:25 /dev/ttyAMA0
crw-rw----    1 root     dialout   204,  65 Feb 17 03:25 /dev/ttyAMA1
crw-rw----    1 root     dialout   204,  66 Feb 17 01:24 /dev/ttyAMA2
```

btuart (as per /lib/systemd/system/hciuart.service) is not running:

```
# ps aux | grep btuart
root       25963  0.0  0.0   3344   520 pts/0    S+   03:28   0:00 grep btuart
```

hcidisabled.service did not run due to "condition not met" (good). hciuart.service was skipped due to "exec-condition" (bad).

```
# systemctl status --no-pager hcidisable hciuart
○ hcidisable.service - Remove HCI kernel driver if WiFi/Bluetooth module is not present
     Loaded: loaded (/usr/lib/systemd/system/hcidisable.service; enabled; preset: enabled)
     Active: inactive (dead)
  Condition: start condition failed at Sat 2024-02-17 01:24:26 UTC; 2h 3min ago
             └─ ConditionPathExists=!/sys/bus/sdio/devices/mmc1:0001:1 was not met

Feb 17 01:24:26 homeassistant systemd[1]: Remove HCI kernel driver if WiFi/Bluetooth module is not present was skipped bec…:0001:1).

○ hciuart.service - Configure Bluetooth Modems connected by UART
     Loaded: loaded (/usr/lib/systemd/system/hciuart.service; enabled; preset: enabled)
     Active: inactive (dead) (Result: exec-condition) since Sat 2024-02-17 01:24:27 UTC; 2h 3min ago
  Condition: start condition failed at Sat 2024-02-17 01:24:26 UTC; 2h 3min ago
    Process: 481 ExecCondition=/bin/sh -c [ "$(cat /proc/device-tree/$(cat /proc/device-tree/aliases/bluetooth)/status)" != "okay" ] (code=exited, status=1/FAILURE)
        CPU: 14ms

Feb 17 01:24:27 homeassistant systemd[1]: Starting Configure Bluetooth Modems connected by UART...
Feb 17 01:24:27 homeassistant systemd[1]: hciuart.service: Skipped due to 'exec-condition'.
Feb 17 01:24:27 homeassistant systemd[1]: Condition check resulted in Configure Bluetooth Modems connected by UART being skipped.
Hint: Some lines were ellipsized, use -l to show in full.
```

Looking at the exec-condition, it check the result of `cat "/proc/device-tree/$(cat /proc/device-tree/aliases/bluetooth)/status"`, which is:

```
# cat "/proc/device-tree/$(cat /proc/device-tree/aliases/bluetooth)/status"
okay#
```

Don't we want `cat "/proc/device-tree/$(cat /proc/device-tree/aliases/bluetooth)/status"` to be **equal** to `okay` instead, though?

Regardless, manually starting btuart seems to get me nowhere. As far as I can tell, it's just exiting.

```
# /usr/bin/btuart
# ps aux | grep btuart
root       29265  0.0  0.0   3344   572 pts/0    S+   03:47   0:00 grep btuart
#
```

**Comment by brunokc** [comment]
Just saw that btuart is a script and it seems to exit early by design on HA Yellow devices?

```
[...]
# Bail out if the kernel is managing the Bluetooth modem initialisation
if ( dmesg | grep -q -E "hci[0-9]+: BCM: chip" ); then
  # On-board bluetooth is already enabled
  exit 0
fi
[...]
```

And we have this:

```
# dmesg | grep -E "hci[0-9]+: BCM: chip"
[    6.645148] Bluetooth: hci0: BCM: chip id 107
#
```

Still, no controllers seem to have been detected:

```
# bluetoothctl list
#
```

**Comment by sairon** [comment]
> Just saw that btuart is a script and it seems to exit early by design on HA Yellow devices?

It's designed to exit early on the whole RPi platform, because the HCI driver is not initialized by the userspace anymore, but by the kernel driver itself early in the boot. In that case, userspace initialization scripts must not be used anymore. You can revert to using them by adding `dtparam=krnbt=off` to `config.txt` in the boot partition, but it is not the default and should work without that. Also make sure there are no other overlays or modifications in your config, i.e. it's the same as [provided in a fresh install](https://github.com/home-assistant/operating-system/blob/dev/buildroot-external/board/raspberrypi/yellow/config.txt).

However, you say it started to misbehave after upgrade from 11.4 to 11.5. There was no change that could have effect on this, so I think that something went wrong only during that particular initialization, what was that could be seen early in `dmesg`. Have you tried whether just simple restart (ideally a cold one, with unplugging the power for a while) doesn't fix that?

**Comment by brunokc** [comment]
@sairon, great analysis! I first checked my config.txt and it was still identical to the fresh install one. Then I went ahead with a cold reboot and sure enough it came back to life. `dmesg` output looks different than when it was erroring out and going to Settings -> Devices shows a working Bluetooth device with live connections.

I'll try to remember this for next time I hit an issue like this during an upgrade.  

Thank you!
